### ISSUE TITLE
ci: go tests failing

### ISSUE THREAD

**Issue by matifali** [coder/coder#7425]
## Problem 
None of the commits to `main` has passed all tests after https://github.com/coder/coder/commit/4e52112112851ada3a589fa5e32e4be5095379eb
total of 9 PRs merged to `main `after this.

## Context: 
I run a script as a cron job to deploy the latest build from a successful PR merge on `main`, as the tests are not being successful since yesterday, I noted a drift in my deployment, and it is not being updated anymore.
I can update my script to deploy on failures, too, as for dogfood we are deploying even if some of the tests fail
The tests that frequently fail are

1. `test-go (ubuntu-latest)`
2. `test-go (macos-latest)`
3. `test-go (windows-2022)`
4. `test-go-psql`

**Comment by johnstcn** [comment]
@matifali Is this a duplicate of https://github.com/coder/coder/issues/7429 ?

**Comment by matifali** [comment]
Yes most probably the reason was that test so we may close it now
